test/test_db/test_db.py

```diff
@@ -22,64 +21,126 @@ def get_connection():
         """Unit tests for custom function duplicate_indicator."""
 
-        # Insert test indicator group
+        # Insert user group
         test_case_name = get_test_case_name()
-        insert_indicator_group_query = f'''INSERT INTO base.indicator_group (name, user_group) VALUES ('{test_case_name}', '{test_case_name}') RETURNING id;'''
-        cursor = self.connection.execute(insert_indicator_group_query)
-        self.connection.commit()
-        indicator_group_id = cursor.fetchone()[0]
+        user_group_id = self.create_user_group(test_case_name)
 
-        # Insert test indicator
-        insert_indicator_query = f'''INSERT INTO base.indicator (name, flag_active, indicator_type_id, indicator_group_id, user_group) VALUES ('{test_case_name}', true, 1, {indicator_group_id}, '{test_case_name}') RETURNING id;'''
-        cursor = self.connection.execute(insert_indicator_query)
-        self.connection.commit()
-        indicator_id = cursor.fetchone()[0]
+        # Insert indicator group
+        indicator_group_id = self.create_indicator_group(test_case_name, user_group_id)
+
+        # Insert indicator
+        indicator_id = self.create_indicator(test_case_name, indicator_group_id, user_group_id)
 
         # Insert test parameter
-        insert_parameter_query = f'''INSERT INTO base.parameter (value, indicator_id, parameter_type_id, user_group) VALUES ('{test_case_name}', {indicator_id}, 1, 'test_group');'''
+        insert_parameter_query = f'''INSERT INTO base.parameter (value, indicator_id, parameter_type_id, user_group_id) VALUES ('{test_case_name}', {indicator_id}, 1, {user_group_id});'''
         self.connection.execute(insert_parameter_query)
-        self.connection.commit()
 
         # Call test duplicate indicator function
         new_test_case_name = get_test_case_name()
         call_test_duplicate_indicator_query = f'''SELECT base.duplicate_indicator({indicator_id}, '{new_test_case_name}');'''
         self.connection.execute(call_test_duplicate_indicator_query)
 
         # Get new indicator and parameter
-        select_new_indicator_query = f'''SELECT a.name, b.value FROM base.indicator a INNER JOIN base.parameter b ON a.id=b.indicator_id WHERE name = '{new_test_case_name}';'''
+        select_new_indicator_query = f'''SELECT a.name, b.value FROM base.indicator a INNER JOIN base.parameter b ON a.id = b.indicator_id WHERE a.name = '{new_test_case_name}';'''
         cursor = self.connection.execute(select_new_indicator_query)
         row = cursor.fetchone()
 
         # Assert duplicated indicator name and parameter value
-        name = row[0]
-        value = row[1]
-        self.assertEqual(name, new_test_case_name)
-        self.assertEqual(value, test_case_name)
+        indiator_name = row[0]
+        parameter_value = row[1]
+        self.assertEqual(indiator_name, new_test_case_name)
+        self.assertEqual(parameter_value, test_case_name)
+
+        # Rollback uncommitted data
+        self.rollback()
```

Comment:
What about putting `self.rollback()` in the `tearDown(self)` method and starting the transaction in the `setUp(self)` method?

Reply:
Also, where do you start the transaction?

Reply:
I do not specifically indicate the beginning of transactions but all the tests run in the same database session opened in setUp.

In the meantime, I have to perform a commit in method test_trigger_update_updated_date in order to validate the test.

This means all insert, update, delete transactions perform in the previous tests during the session would be committed once I execute this commit. So I execute the rollback at the end of each test rather than in tearDown to avoid that.